### PR TITLE
Objc to swift conversion

### PR DESCRIPTION
**Pull request by stury**
The prior SPM support that was added, didn't work for Swift apps when consuming the full SDK. I've simplified the Swift Package, and converted all of the code to Swift, without any breaking changes.  

All Unit Tests run successfully!

Obj-C consumers of this package may have to switch to using the `@import TDOAuth;` notation to import the header, since there is no longer a `TDOAuth.h` file in the repo anymore. All other aspects of Objective-C consumption are the same as they were. 

I've marked all of the `TDOAuth` public methods as `deprecated`, so we can remove these in the next major version release.

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: Source/compat/Compat.swift [hunk 1]
```diff
@@ -20,7 +16,7 @@ import TDOAuthSwift
                                   consumerSecret: String,
                                   accessToken: String?,
                                   tokenSecret: String?,
-                                  signatureMethod: TDOAuthSignatureMethod) -> URLRequest? {
+                                  signatureMethod: TDOAuthSignatureMethod) -> URLRequest! {
```
Comment (master-nevi) [inline Source/compat/Compat.swift:19]: did this return value have to change? Is this not considered a breaking change?

File: TDOAuth.podspec [hunk 2]
```diff
@@ -25,17 +25,13 @@ Pod::Spec.new do |s|
...
 
-  s.swift_versions = [ '4.0', '4.2', '5.0' ]
+  s.swift_versions = [ '4.0', '4.2', '5.0', '5.1', '5.2', '5.3', '5.4', '5.5', '5.6' ]
 
-  s.default_subspec = 'ObjC'
+  s.default_subspec = 'Swift'
 
   s.subspec 'Swift' do |ss|
-    ss.source_files = 'Source/*.swift'
+    ss.source_files = 'Source/*.swift', 'Source/compat/*.{swift}'
```
Comment (master-nevi) [inline TDOAuth.podspec:33]: Should this be all inclusive and include TDOAuth1.swift ?
```suggestion
    ss.source_files = 'Source/*.swift', 'Source/**/*.{swift}'
```

File: Source/compat/Compat.swift (hunk 1 above)
Comment (stury) [reply Source/compat/Compat.swift:19]: The original Objective-C interfaces returned `URLRequest!` when used in Swift, so I'm just making sure they are the same as would originally have been used.

**Review by master-nevi — APPROVED** [review]
On the surface it LGTM. Might want to get a +1 from one of the original maintainers.

**Pull request merged**